app/javascript/components/axis.js

```diff
@@ -0,0 +1,175 @@
+    this._xOffset = dataset.xOffset
+    this._y0 = dataset.yOffset
+    this.hexDiameter = dataset.hexDiameter
+    this.mapWidth = dataset.mapWidth
+    this.mapHeight = dataset.mapHeight
+    this.dashFilledLength = dataset.dashFilledLength
+    this.dashBlankLength = dataset.dashBlankLength
+    this.color = dataset.color
+  }
+
+  get dash() {
+    return [this.dashFilledLength, this.dashBlankLength]
+  }
+
+  drawLines(ctx) {
+    var shifted
+    var x0 = this._x0
+    var y0 = this._y0
+
+    // Draw first set of lines
+    this._drawDashedLines(ctx, x0, y0)
+
+    for (var i = this.shifts; i > 0; i--) {
+      // Shift set of offsets to next set of offsets
+      shifted = this.shift(x0, y0)
+      x0 = shifted.x0
+      y0 = shifted.y0
+
+      // Draw shifted lines
+      this._drawDashedLines(ctx, x0, y0)
+    }
+  }
+
+  get m() {
+    return Math.tan(this.radians)
+  }
+
+  shift(x0, y0) {
+    if (this.radians === 0 || this.radians === Math.PI) {
+      // Shift horizontal lines
+      x0 = x0 + this.dashFilledLength * 1.5
+      y0 = y0 + this.hexDiameter / 2
+    } else if (this.radians === Math.PI / 2 || this.radians === 3 * Math.PI / 2) {
+      // Shift vertical lines
+      console.log("Not implemented")
+    } else {
+      // Shift angled lines
+      x0 = x0 + Math.cos(this.radians) * this.dashFilledLength * 3
+      y0 = y0 + Math.sin(this.radians) * this.dashFilledLength
+    }
+
+    return { x0, y0 }
+  }
+
+  get shifts() {
+    if (this.radians === 0 || this.radians === Math.PI) {
+      // Horizontal
+      return 1
+    } else if (this.radians === Math.PI / 2 || this.radians === 3 * Math.PI / 2) {
+      // vertical lines
+      console.log("Not implemented")
+    } else {
+      // Angled
+      return 2
+    }
+  }
+
+  get radians() {
+    return this._radians(this.degrees)
+  }
+
+  get xStep() {
+    if (this.radians === 0 || this.radians === Math.PI) {
+      // Horizontal lines
+      return 0
+    } else if (this.radians === Math.PI / 2 || this.radians === 3 * Math.PI / 2) {
+      // vertical lines
+      console.log("Not implemented")
+    } else {
+      // Angled lines
+      return this.hexDiameter
+    }
+  }
+
+  get yStep() {
+    if (this.radians === 0 || this.radians === Math.PI) {
+      // Horizontal Lines
+      return this.hexDiameter
+    } else if (this.radians === Math.PI / 2 || this.radians === 3 * Math.PI / 2) {
+      // vertical lines
+      console.log("Not implemented")
+    } else {
+      // Angled Lines
+      return 0
+    }
+  }
+
+  _drawDashedLines(ctx, x0, y0) {
+    var line
+    var xStep = this.xStep
+    var yStep = this.yStep
+
+    ctx.strokeStyle = this.color
+    ctx.setLineDash(this.dash)
+    ctx.beginPath()
+    for (var xi = x0, yi = y0; xi <= this.mapWidth + this._xBuffer && yi <= this.mapHeight; xi += this.xStep, yi += this.yStep) {
+      line = this._getLineCoordinates(xi, yi)
+      ctx.moveTo(line.x1, line.y1);
+      ctx.lineTo(line.x2, line.y2);
```

Comment:
@schmidglin , when commenting on code, you can click the little `+` sign next to the line number to add a comment to the line. That seems to be a great way to attach comments to the code.

For example:
@manik1235 , don't forget, we're not doing semicolons in our JavaScript style! Will you please remove them?